Jenkinsfile

```diff
@@ -116,10 +166,21 @@ pipeline {
                 )
             }
         }
+        /*
+          Deploy stage will only be executed for deployable branches. These include master and any patch branch matching M.m.x format (i.e. 2.10.x, 2.9.x, etc...).
+          It will also only deploy in the presence of an environment variable JENKINS_ENV = 'prod'. This can be passed in globally from the jenkins master node settings.
```

Comment:
❓ Curious to know, when do we have a deployable branch that matches the pattern but shouldn't be deployed? Couldn't we always use the `JENKINS_ENV` environment variable and not use the pattern match?

Reply:
We could, only the plan was to set that env variable globally per jenkins master instance. This would allow us to test pipelines while auto-skipping the deploy phase when it is run on the test jenkins environment, vs the prod jenkins environment. 

As for the deployable branches. The idea here is that we want to avoid building and deploying on branches unless they are going to have unique versions in their poms (which the 2.10.x, 2.9.x, etc...) branches do. All work is currently done using forks, however that doesn't necessarily stop anyone with commit privileges from pushing an arbitrary branch to this repo. Lets say for example sake that it was branched from master (currently at 2.11.0-SNAPSHOT) and the version number wasn't changed, without this check jenkins would automatically discover, build, and deploy it if a jenkinsfile was present. This check allows for jenkins to discover and build it, but to skip the deploy step. 

The benefit of doing it this way is that, 
* local branches with potentially conflicting versions don't cause issues
* patch fix branches can be built, verified, and deployed based on scm changes
* all deploy steps can be disabled on a per jenkins master basis